dace/codegen/targets/common.py

```diff
@@ -38,8 +39,8 @@ def sym2cpp(s, arrayexprs: Optional[Set[str]] = None) -> Union[str, List[str]]:
     :return: C++-compilable expression or list thereof.
     """
     if not isinstance(s, list):
-        return cppunparse.pyexpr2cpp(symbolic.symstr(s, arrayexprs))
-    return [cppunparse.pyexpr2cpp(symbolic.symstr(d, arrayexprs)) for d in s]
+        return symbolic.symstr(s, arrayexprs)
+    return [symbolic.symstr(d, arrayexprs) for d in s]
```

Comment:
I'm a bit afraid of this change

Reply:
Well, there is no way to make this work otherwise. `symstr` return cpp strings and it is too deep to untangle (even uses a custom sympy printer to output cpp code). `symstr` also fixes min, max, floor, ceil, etc., so we cannot avoid using it. With the methods that we have now, this is the only way that produces cpp code with the above methods handled properly. The original code, took symbolic expressions, used `symstr` to make them (cpp) code and then tried to parse them with Python ast module. This fails whenever the generated cpp code is not valid Python code.